src/widgets/dock_area/mod.rs

```diff
@@ -348,6 +365,79 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
                 }
             }
         }
+
+        separators
+    }
+
+    fn show_cross_section_separators(&mut self, ui: &mut Ui, separators: Vec<LineSeparator>) {
+        let style = self.style.as_ref().unwrap();
+        let mut cross_section_seperators: Vec<CrossSeparator> = vec![];
+
+        // detect overlapping line separators
+        for (i, sep1) in separators.iter().enumerate() {
+            for sep2 in separators[i + 1..].iter() {
```

Comment:
Shouldn't there be an `if i == separators.len() - 1 { break; }`? That `i + 1` worries me because it will go out of bounds on the final element.